packages/uploadthing/src/client.ts

```diff
@@ -122,6 +129,20 @@ export const genUploader = <TRouter extends FileRouter>(
     );
 };
 
+type Done = { status: "done"; callbackData: unknown };
+type StillWaiting = { status: "still waiting" };
+type PollingResponse = Done | StillWaiting;
+
+const isPollingResponse = (input: unknown): input is PollingResponse => {
+  if (!isObject(input)) return false;
+  if (input.status === "done") return "callbackData" in input;
+  return input.status === "still waiting";
+};
+
+const isPollDone: Predicate.Refinement<PollingResponse, Done> = (
+  input,
+): input is Done => input.status === "done";
```

Comment:
why does this function need 2 explicit types? will it be "fixed" in TS5.5?

Reply:
no idea, refinements alway needed convincing. I guess we can remove `Predicate.Refinement<PollingResponse, Done>` and let it just infer the refinement. Either is fine